core/client/db/src/lib.rs

```diff
@@ -580,7 +580,8 @@ impl<Block> client::backend::Backend<Block, Blake2Hasher, RlpCodec> for Backend<
 				}
 			}
 			let number_u64 = number.as_();
-			let commit = self.storage.state_db.insert_block(&hash, number_u64, &pending_block.header.parent_hash(), changeset);
+			let commit = self.storage.state_db.insert_block(&hash, number_u64, &pending_block.header.parent_hash(), changeset)
+				.map_err(|e: state_db::Error<io::Error>| client::error::Error::from(format!("State database error: {:?}", e)))?;
```

Comment:
perhaps add a special variant for State DB errors to the `ClientError` type? I don't know what variant is being produced with a regular `From` implementation.